Sources/PointFree/Routes.swift

```diff
@@ -91,28 +92,29 @@ private func redirectUnrelatedDomains<A>(
         } else {
           return nil
         }
-        }
-        ?? middleware(conn)
+        } ?? middleware(conn)
     }
 }
 
-private let httpAllowedHosts = [
+private let allowedInsecureHosts: [String] = [
   "127.0.0.1",
   "0.0.0.0",
   "localhost"
 ]
 
 // TODO: move to HttpPipeline
-private func requireHttps<A>(
+private func requireHerokuHttps<A>(
```

Comment:
if we move this to `HttpPipeline` I think it could be made into general redirect middleware that takes a predicate `(URLRequest) -> Bool` to determine if to redirect, and a request rewriting function `(URLRequest) -> URL` for the redirect. it should also handle any non-`GET` requests to redirect to a 400 page.